src/app/server/TermsAndConditionsProvider.h

```diff
@@ -0,0 +1,78 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <stdint.h>
+
+#include <lib/core/CHIPConfig.h>
+#include <lib/core/CHIPError.h>
+
+namespace chip {
+namespace app {
+
+/**
+ * @brief Data access layer for the required terms and conditions and the store for the user acceptance.
+ */
+class TermsAndConditionsProvider
```

Comment:
Not saying you need to change it now, but would TermsAndConditionsStorageProvider be a better name?

Reply:
`Provider` is the more common suffix in the code base already. Since this is an true interface (abstract base class), we should not assume "storage" as in assuming this code has to use PersistentStorageDelegate or such.